packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneAttachmentBlobs.spec.ts

```diff
@@ -268,6 +268,80 @@ describeNoCompat("GC attachment blob tombstone tests", (getTestObjectProvider) =
+         */
+        async function assertDoesNotRejectInverse(block: Promise<any>, message?: string | Error) {
+            return assert.rejects(block, message);
+        };
+
+        /**
+         * This test validates that when blobs are de-duped in different containers, these containers can use these
+         * blobs irrespective of whether the original blob is tombstoned. Basically, after uploading a blob, a container
+         * should be able to use it the same way whether it was de-duped or not.
+         * Note: Currently this fails because the behavior is not as expected and so it's expecting an error.
+         */
+        itExpects("should allow access to blobs that are de-duped in different containers",
+        [
+            { eventName: "fluid:telemetry:BlobManager:GC_Tombstone_Blob_Requested" }
+        ],
+        async () => {
+            const { dataStore: mainDataStore, summarizer } = await createDataStoreAndSummarizer();
+
+            // Upload an attachment blob.
+            const blobContents = "Blob contents";
+            const blobHandle = await mainDataStore._runtime.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+
+            // Reference and then unreference the blob so that it's unreferenced in the next summary.
+            mainDataStore._root.set("blob1", blobHandle);
+            mainDataStore._root.delete("blob1");
+
+            // Summarize so that the above attachment blobs are marked unreferenced.
+            await provider.ensureSynchronized();
+            const summary1 = await summarizeNow(summarizer);
+
+            // Wait for half sweep timeout and load a container. This container will upload a blob with the same content
+            // as above so that it is de-duped. This container should be able to use this blob until its session
+            // expires.
+            await delay(sweepTimeoutMs / 2);
+            const container2 = await loadContainer(summary1.summaryVersion);
+            const container2MainDataStore = await requestFluidObject<ITestDataObject>(container2, "default");
+            // Upload the blob and keep the handle around until the blob uploaded by first container is tombstoned.
+            const container2BlobHandle =
+                await container2MainDataStore._runtime.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+
+            // Wait for sweep timeout so that the blob uploaded by the first container is tombstoned.
+            await delay(sweepTimeoutMs + 10);
+
+            // Send an op to update the current reference timestamp that GC uses to make sweep ready objects.
+            mainDataStore._root.set("key", "value");
+            await provider.ensureSynchronized();
+
+            // Summarize so that the tombstoned blob is now part of the summary.
+            const summary2 = await summarizeNow(summarizer);
+
+            // Load a container from this summary and upload a blob with the same content as the tombstoned blob.
```

Comment:
So this is the 3rd copy of the blob?  Why is this necessary?

Reply:
In container2, the blob is not tombstoned but in container3 it is tombstoned. It tests 2 different scenarios:
1. Container3 - The blob is tombstoned and can't even be retrieved.
2. Container2- The blob in not tombstoned. It can be retrieved but referencing it will result in an error in other containers.